src/services/item/plugins/action/service.ts

```diff
@@ -76,9 +85,44 @@ export class ActionItemService {
     return repositories.actionRepository.getForItem(item.path, {
       sampleSize: size,
       view,
+      memberId: permission === PermissionLevel.Admin ? undefined : actor.id,
```

Comment:
I think there's a major problem here because you cannot handle `public` data (empty user both means public and admin).

Reply:
We didn't define the public case together, so either:
- if you don't have time: add `preHandler: fastify.verifyAuthentication` to your endpoints (you can have a look to any endpoint in the codebase). It will ensure `actor` is always defined. But you will need to add the following code in your service so actor has the correct type.
```
 if (!actor) {
      throw new UnauthorizedMember(actor);
    }
```
- you have time: I think that a `public` item should be able to display the aggregated data only. And it shouldn't be possible to get individual actions only if you're signed in.

Reply:
I try to fix it and make it clear. For the individual actions, the workflow is in the following:

1. Check if the user is logged in
2. Check the right through `this.itemService.get(..., PermissionLevel.Read,)`. If it returns, the user has at least read access to the item or the item is public.
3.  Get the user's permission through `repositories.itemMembershipRepository.getInherited(...)`.
4.  If the permission is Admin, get all actions. Otherwise, get actions created by the actor.

In this case, for a public item, unauthorized members cannot access it, and authorized members can get the actions created by themselves even if they don't have any permission. Does that make sense?

For the aggregated data, we only check the right through `this.itemService.get(..., PermissionLevel.Read,)`. If it returns, we return the aggregated data.